website/server/controllers/api-v3/chat.js

```diff
@@ -100,6 +100,32 @@ api.postChat = {
 
     let group = await Group.getGroup({user, groupId});
 
+    //////////////////////////////////////////////////////////////
+    // Initial code to check for slurs and revoke chat priviliges  
+
+    // Move to a checkForSlurs function?
+      
+    // Temporary list of fake slurs. Somehow this needs to be loaded in or passed
+    let slurList = ["kicking puppies",  "mean things"];
+
+    // Replace all punctuation with spaces to make for an easier search. Also add spaces before and after
+    // the test message (so we can search for slurs surrounded by spaces, preventing partial word matches)
+    // This is not an elegant way to do this, I'm sure
+    let message = req.body.message;
+    let noPunctMessage = " " + message.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g," ") + " ";
+
+    // Check for any slurs in the message
+    for (let i=0; i<slurList.length; i++){
+      let tmp = noPunctMessage.search(" " + slurList[i] + " "); // Don't match partial words
+      if (tmp > -1) {
+        // Need a way to make revoking chat permanent
+        user.muteUser(message, groupId);
+        throw new NotFound('Your message contained inapropriate language, and your chat privileges have been revoked.');
```

Comment:
This should be moved to a local. Use `res.t()`.

Also, you should move this check below, because you need to check the privacy setting of the group. Private groups can use banned language.